### PR TITLE
PPC Simple Product with Custom Options Support

### PR DESCRIPTION
**Pull request by mccottry**
# Description
PPC Simple Product with Custom Options Support

Fixes: https://app.asana.com/0/941895179897714/1161793187073243

#changelog Product Page Checkout - Simple Products with Custom Options support

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml [hunk 1]
```diff
@@ -61,18 +69,37 @@ var boltConfigPDP = {
...
+                    if (advice == null) {
+                        advice = Validation.createAdvice(name, fileInput, false, '<?php echo $this->__('Files not supported with Product Page BoltCheckout. Please add product to cart and then use Bolt from the cart page to checkout.'); ?>');
+                    }
+                    Validation.showAdvice(fileInput, advice, name);
+                    fileInput.focus();
+                    throw '<?php echo $this->__('Files not supported with Product Page BoltCheckout. Please add product to cart and then use Bolt from the cart page to checkout.'); ?>';
+                }
+            });
+        } catch (e) {
+            if ((typeof BoltPopup !== 'undefined' && typeof e === 'string')) {
```
Comment (adentong) [inline app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: Tiny nit: redundant pair of brackets

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml [hunk 2]
```diff
@@ -61,18 +69,37 @@ var boltConfigPDP = {
...
             }
             return false;
         }
-        return new Validation('product_addtocart_form').validate();
+        return formValidation.validate();
     },
     /**
      * Returns cart data provided to BoltConnect.configureProductCheckout
-     * @returns {{total: number, items: ([{reference: number, quantity: number, price: number, name: string, options: string}]|[])}}
+     * @returns {{currency: string, items: ([{reference: number, quantity: number, price: number, name: string, options: string}]|[])}}
```
Comment (adentong) [inline app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: Trying to understand this. Why was total removed?

File: app/design/frontend/base/default/template/boltpay/connectjs.phtml [hunk 3]
```diff
@@ -22,6 +22,35 @@ $isAllowedConnectJs = $this->isAllowedConnectJsOnCurrentPage();
 <?php if ($isAllowedConnectJs): ?>
     <?php $publishableKey = $this->getPublishableKeyForRoute(); ?>
     <?php if ($publishableKey): ?>
+        <script>
```
Comment (adentong) [inline app/design/frontend/base/default/template/boltpay/connectjs.phtml]: This seems kinda unrelated to PPC? What does this do?

Comment (mccottry) [reply app/design/frontend/base/default/template/boltpay/connectjs.phtml]: @adentong This is related and required for PPC.  I've updated the comment below to be more clear.

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml (hunk 2 above)
Comment (mccottry) [reply app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: It is not required.  It should be a calculated value.  It is not spec'd, and also most alarmingly, core in SF seems to be changing this.  This has changed from a JS number to requiring a JS string.  We don't want to rely on this value that is unnecessary and removing it is easier and more reliable than having PPC break due to core changes.

Spec:
https://docs.bolt.com/docs/54-product-checkout

New error being raised:
image

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml (hunk 1 above)
Comment (mccottry) [reply app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: yep.

File: app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml (hunk 2 above)
Comment (adentong) [reply app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml]: Got it.

**Comment by adentong** [comment]
@johnmccombs1 @ryan-timothy-albert @matt-thomason @Jiyuan-Bolt Let's try to get another review on this so it can be merged.
